### PR TITLE
improve docs, examples and tests

### PR DESCRIPTION
**Pull request by dmitrykruglov**

### CODE REVIEW

**Review by benoittoulme — APPROVED** [review]

File: provider/provider_test.go [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+}
+
+func testAccPreCheck(t *testing.T) {
+	// You can add code here to run prior to any test case execution, for example assertions
+	// about the appropriate environment variables being set are common to see in a pre-check
+	// function.
+}
+
+func TestProvider(t *testing.T) {
+	if err := Provider().InternalValidate(); err != nil {
```
Comment (masha1997) [inline provider/provider_test.go:29]: nice :)

**Review by masha1997 — APPROVED** [review]

File: docs/resources/itsi_kpi_threshold_template.md [hunk 2]
```diff
@@ -3,14 +3,123 @@
...
 
-
-
-
+Manages a KPI threshold template.
+
+## Example Usage
+
+```terraform
+resource "itsi_kpi_threshold_template" "example" {
```
Comment (masha1997) [inline docs/resources/itsi_kpi_threshold_template.md:16]: nothing required to be changed but here could be mentioned that _time-variate static_ threshold could be supported only via template like:
policy_type = "static"
and then dynamic_param is force set to 0

**Comment by masha1997** [comment]
Thank you! looks super

File: docs/resources/itsi_kpi_threshold_template.md (hunk 2 above)
Comment (dmitrykruglov) [reply docs/resources/itsi_kpi_threshold_template.md]: Thanks! I added an example for a static threshold template.

**Review by masha1997 — APPROVED** [review]

**Pull request merged**